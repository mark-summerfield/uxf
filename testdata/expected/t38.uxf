uxf 1.0
=IntPair First:int Second:int
=IP4 A:int B:int C:int D:int
=RGB Red:int Green:int Blue:int
[#<Unused TTypes produce warnings when converting from uxf to uxf.
One way to avoid the warnings is to use them for empty tables.> 
  (IntPair)
  (IP4
    168 1 4 29
    192 0 18 27
  )
  (RGB)
]
