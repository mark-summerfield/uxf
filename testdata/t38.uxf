uxf 1.0
= IP4 A:int B:int C:int D:int
= RGB Red:int Green:int Blue:int
= IntPair First:int Second:int
[ #<Unused TTypes are dropped when converting from uxf to uxf;
    to preserve them, just use them to make an empty table>
  (IntPair)
  (IP4 168 1 4 29 192 0 18 27)
  (RGB)
]
