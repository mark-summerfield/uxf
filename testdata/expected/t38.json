{
  "UXF^ttypes": [
    {
      "UXF^ttype": {
        "name": "IP4",
        "fields": {
          "A": "int",
          "B": "int",
          "C": "int",
          "D": "int"
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "IntPair",
        "fields": {
          "First": "int",
          "Second": "int"
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "RGB",
        "fields": {
          "Red": "int",
          "Green": "int",
          "Blue": "int"
        }
      }
    }
  ],
  "UXF^data": {
    "UXF^list": {
      "comment": "Unused TTypes produce warnings when converting from uxf to uxf.\nOne way to avoid the warnings is to use them for empty tables.",
      "vtype": null,
      "list": [
        {
          "UXF^table": {
            "name": "IntPair",
            "comment": null,
            "fields": {
              "First": "int",
              "Second": "int"
            },
            "records": []
          }
        },
        {
          "UXF^table": {
            "name": "IP4",
            "comment": null,
            "fields": {
              "A": "int",
              "B": "int",
              "C": "int",
              "D": "int"
            },
            "records": [
              [
                168,
                1,
                4,
                29
              ],
              [
                192,
                0,
                18,
                27
              ]
            ]
          }
        },
        {
          "UXF^table": {
            "name": "RGB",
            "comment": null,
            "fields": {
              "Red": "int",
              "Green": "int",
              "Blue": "int"
            },
            "records": []
          }
        }
      ]
    }
  }
}